### DISCUSSION TITLE
Pass an event to handleSubmit?

### DISCUSSION THREAD

**Discussion by markwoon** [react-hook-form/react-hook-form#6763] [Q&A]
I have to use an onclick handler to do some work before calling `handleSubmit`.

Is there a way to pass the event on to `handleSubmit`?

**Comment by bluebill1049** [comment]
It's the second argument for the callback

**Reply by markwoon** [reply]
Apologies for not being clearer.  I am not looking for how to get the event in the callback, I'm looking to pass the event to `handleSubmit()` if I call it directly.

```
const myHandler = (data, event) => {
  // do something
};


const preHandler = (event) => {
  // how do I pass this event on so that myHandler will get it?
  handleSubmit(myHandler)();
}
```

**Reply by dmlsrc** [reply]
In case anyone lands here from Google:

```
const onSubmitSuccess = (submitData, e) => {
};

const onSubmitError = (submitErrors, e) => {
};

const onNextClick = (e) => {
    handleSubmit(onSubmitSuccess, onSubmitError)(e);
};

<button type="submit" onClick={onNextClick}>Next</button>

```
